feat(dashboard): update display of tracking/label URLs on order details

docloulou opened the pull request:
### Overview

This PR updates the fulfillment label rendering by replacing the deprecated .url field with the new tracking_url and label_url fields, following the MedusaJS FulfillmentLabel model.

### Changes

- Implemented conditional rendering for both tracking_url and label_url.
- If a tracking_url exists, it is displayed as a clickable link.
- Similarly, if a label_url exists, it is displayed as a clickable link.
- If neither link is provided, only the tracking number is shown.
- Maintained existing styling for interactive elements.

![image](https://github.com/user-attachments/assets/187bd8d5-d6c2-4f06-8f61-f9bb7d02da11)

### Reference

This update is based on the MedusaJS FulfillmentLabel model:
https://docs.medusajs.com/resources/references/fulfillment/models/FulfillmentLabel

Please review the changes and provide feedback for further improvements.

docloulou committed:
feat(fulfillment): update display of tracking and label URLs on order details

Adapt component to use tracking_url and label_url fields based on the MedusaJS FulfillmentLabel model.
Reference: https://docs.medusajs.com/resources/references/fulfillment/models/FulfillmentLabel

docloulou committed:
fix UI

docloulou commented:
@riqwan @adrien2p sorry to bother you, could this PR be reopened/reviewed please? i need it in my project (to avoid a custom widget)

willbouch commented on packages/admin/dashboard/src/routes/orders/order-detail/components/order-fulfillment-section/order-fulfillment-section.tsx:
```diff
@@ -405,33 +405,54 @@ const Fulfillment = ({
+                  tlink.tracking_url &&
+                  tlink.tracking_url.length > 0 &&
+                  tlink.tracking_url !== "#";
+                const hasLabelUrl =
```
we could write a small little function here to avoid duplicating the logic

docloulou replied:
done

willbouch reviewed (commented):
Thanks for the contribution! Left a small comment. Can you also push a  changeset

docloulou committed:
Merge branch 'medusajs:develop' into patch-3

docloulou committed:
refactor(order-fulfillment): simplify URL validation logic

- Introduced a helper function `isValidUrl` to streamline the validation of tracking and label URLs.
- Reduced code duplication by replacing individual URL checks with the new helper function.
- Improved readability and maintainability of the fulfillment component.

No functional changes were made; this is purely a code cleanup.

docloulou committed:
changeset

docloulou commented:
> Thanks for the contribution! Left a small comment. Can you also push a changeset

done :)

willbouch committed:
Create pink-humans-thank.md

willbouch commented:
I wanted to update the changeset because the package you are trying to change is `dashboard` and not fulfillment but it ended up creating another changeset. Are you able to delete it ? Sorry haha

willbouch commented on packages/admin/dashboard/src/routes/orders/order-detail/components/order-fulfillment-section/order-fulfillment-section.tsx:
```diff
@@ -408,22 +408,40 @@ const Fulfillment = ({
               {fulfillment.labels.map((tlink) => {
-                const hasUrl =
-                  tlink.url && tlink.url.length > 0 && tlink.url !== "#"
+                const isValidUrl = (url?: string) =>
```
I feel like it is a waste to declare that function on every iteration of the loop don't you think? We could pull it out of the .map

adrien2p reviewed (commented):
Thanks for the contribution 👌Just william comment and if you could remove one of the changeset please

docloulou committed:
Delete .changeset/neat-ducks-shave.md

docloulou committed:
Moved the `isValidUrl` function to a higher scope for better accessibility and reduced redundancy.

docloulou commented:
done
